File: c++/programaUno.cpp
/*
    Desarrolla un programa que crea un vector de longitud N con numeros enteros positivos 
    y negativos, apartir de este vector generar un nuevo vector binario, donde los numeros negativos sean 0
    y los positivos sean 1;
*/

#include <iostream>
using namespace std;

int main()
{
    int size;
    cout<<"Ingresa la dimension del arreglo: ";
    cin >>size;
    int array[size];
    int arrayBinario[size];
    for ( int i = 0; i < size; i++){
        cout<<"Ingresa el valor de la posicion ["<<i<<"]: ";
        cin>>array[i];
        if (array[i] >= 0){
            arrayBinario[i] = 1;
        }else{
            arrayBinario[i] = 0;
        }
    }
    cout<<"Array Original."<<endl;
    for( int i = 0; i < size; i++){
        cout<<array[i];
    }

    cout<<"\nArray Binario"<<endl;
    for( int i = 0; i < size; i++){
        cout<<arrayBinario[i];
    }
    cout<<"\n";

    return 0;
}

File: java/programaUno.java
/*
    Desarrolla un programa que crea un vector de longitud N con numeros enteros positivos 
    y negativos, apartir de este vector generar un nuevo vector binario, donde los numeros negativos sean 0
    y los positivos sean 1;
*/
import java.util.Scanner;

class ProgramaUno {

    public static void main(String[] args){
        Scanner scan = new Scanner(System.in);
        System.out.print("Ingresa las dimensiones del array: ");
        int size = scan.nextInt();
        int array[] = new int[size];
        int arrayBinario[] = new int[size]; 

        for( int i = 0; i < array.length; i++ ){
            System.out.print("Ingresa el valor ["+i+"]: ");
            array[i] = scan.nextInt();
            if( array[i] >= 0){
                arrayBinario[i] = 1;
            }  else{
                arrayBinario[i] = 0;
            }
        }
        for(int i = 0; i< arrayBinario.length; i++){
            System.out.println(arrayBinario[i]);
        }
    }
}